graph-builder/src/graph.rs

```diff
@@ -249,21 +255,27 @@ pub fn run<'a>(settings: &'a config::AppSettings, state: &State) -> ! {
             }
         };
 
-        match serde_json::to_string(&graph) {
-            Ok(json) => {
-                *state.json.write() = json;
+        let json_graph = match serde_json::to_string(&graph) {
+            Ok(json) => json,
+            Err(err) => {
+                error!("Failed to serialize graph: {}", err);
+                continue;
+            }
+        };
 
-                if first_success {
-                    *state.ready.write() = true;
-                    first_success = false;
-                };
+        *state.json.write() = json_graph;
 
-                let nodes_count = graph.releases_count();
-                GRAPH_FINAL_RELEASES.set(nodes_count as i64);
-                debug!("graph update completed, {} valid releases", nodes_count);
-            }
-            Err(err) => error!("Failed to serialize graph: {}", err),
+        if first_success {
+            *state.ready.write() = true;
+            first_success = false;
         };
+
+        let last_refresh = chrono::Utc::now();
+        GRAPH_LAST_SUCCESSFUL_REFRESH.set(last_refresh.timestamp() as i64);
+
+        let nodes_count = graph.releases_count();
```

Comment:
I think you could use this directly in the next statement

Reply:
No, this is also used by the `debug!` one line further below.

Reply:
You could read `GRAPH_FINAL_RELEASES` back, but I'm also fine keeping the variable. Just out of curiosity, can you think of a situation where that global value would be different from the global variable?

Reply:
Fair point. I think I'm generally approaching metrics as write-only for the sake of logic separation, as the values flow from our logic to the metrics and not viceversa.
The only real difference is that those have different types (unsigned vs signed), but I think I'm more concerned over keeping metrics as write-only over possible sign-conversion issues.